flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/tasks/CheckpointCoordinatorConfiguration.java

```diff
@@ -36,6 +38,8 @@
 
 	private static final long serialVersionUID = 2L;
 
+	private String stateBackendName;
```

Comment:
this doesn't seem like the right place to put it; we are polluting a runtime configuration class with information that is only relevant to the REST API. This class should only contain things that are relevant for the CheckpointCoordinator; that's this class purpose after all.

Reply:
I think it's the name here that makes people confused. the `stateBackendName` here is `checkpoint backend name`, such as `MemoryStateBackend`, `FsStateBackend` or `RocksDBStateBackend`, and I think they'are relevant with `CheckpointCoordinator`, we'll also pass the backend into [CheckpointCoordinator](https://github.com/apache/flink/blob/8c6cc4505a4c27daadb00cd94df8a7e955eb8d52/flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java#L231) to init the checkpoint directory.
Maybe we could update the name here, what do you think?

Reply:
What functional purpose would this field have for the coordinator? It already has the _actual backend_.

Reply:
Ahh, got it.
Maybe we should get the `stateBackendName` from `AccessExecutionGraph` directly(add a new function `getStateBackend` in `AccessExecutionGraph`, pass the `stateBackendName` into `ArchivedExecutionGraph` when init a new `ArchivedExecutionGraph`), what do you think?

Reply:
That's a _better_ option.

I would suggest adding `CheckpointCoordinator#getStateBackendName()` that is called in `ArchivedExecutionGraph#from()`. Your plan should cover the rest.